apps/src/code-studio/initApp/project.js

```diff
@@ -1193,6 +1194,17 @@ var projects = module.exports = {
    */
   load() {
     var deferred = new $.Deferred();
+
+    // Use the sources-public API for dancelab shares. Responses from this API
+    // can be publicly cached, which is helpful for HoC scalability in the
+    // celebrity tweet scenario where a single share link gets many hits.
+    const useSourcesPublic = appOptions.share && appOptions.level &&
+      appOptions.level.projectType === 'dance';
```

Comment:
I'd love to see us do this for all share pages, not just dance.  I assume we're just being super-careful here by isolating this change to one app type?

Reply:
Yes, the goal here is to limit this risk of disruption in classrooms, while maximizing the benefit for HoC. One could argue that any alleviated pressure on S3 would be a benefit for HoC, though. 

One scenario where I could see this being a pain would be if someone is updating an app (applab, let's say) on their computer while testing it out on their phone. There may be other scenarios we haven't though of, too. So I think I'm inclined to leave this as-is for now and consider broadening to other app types post-hoc.